src/NUnitEngine/nunit.engine.core/Agents/RemoteTestAgent.cs

```diff
@@ -107,7 +121,7 @@ public bool Start()
             return true;
         }
 
-        public override void Stop()
+        public void ShutDown()
```

Comment:
Agents used `Stop` to indicate a distinction with `ShutDown`, which has a special meaning for Services. Services youse `StartUp` and `ShutDown`, whereas communications elements have used `Start` and `Stop`. Where a comm element is also a service, `Start` and `Stop` are typically called as part of `StartUp` and `ShutDown`. Having these as separate methods makes some unit-testing easier.

Reply:
I will say, I've never been aware of that design intent! 😅

I have no strong opinion on the final naming. Joseph - re: Charlie's comment about unit testing, do you think that will cause an issue, based on your final goal here?

Reply:
@ChrisMaddock Well, to put it differently, the intent is seen in the definition of `IService`, which has `StartUp` and `Shutdown` methods. There is no rule against using the same method names outside of the interface, but it's potentially confusing... or maybe only confusing to me. :smile: